docs/modules/clients/pages/java-thin-client.adoc

```diff
@@ -0,0 +1,1126 @@
+To fulfill this requirement, you can configure the {client-name} to use only defined outbound ports.
+
+Example:
+
+[source,java]
+----
+NetworkConfig networkConfig = config.getNetworkConfig();
+// ports between 34700 and 34710
+networkConfig.addOutboundPortDefinition("34700-34710");
+// comma separated ports
+networkConfig.addOutboundPortDefinition("34700,34701,34702,34703");
+networkConfig.addOutboundPort(34705);
+----
+
+[[java-client-operation-modes]]
+==== Setting Smart Routing
+
+Hazelcast clients have two operation modes because of the distributed nature of the data and cluster.
+
+**Smart Client**: In the Smart mode, the clients connect to each cluster member.
+Since each xref:overview:data-partitioning.adoc[data partition] uses the well known and consistent hashing algorithm,
+each client can send an operation to the cluster member that owns the partition that holds their data,
+which increases the overall throughput and efficiency.
+
+NOTE: The **Smart Client** mode is currently not implemented in the {client-name}. Support for Smart Client will be added in the future versions.
+
+**Unisocket Client**:  In the Unisocket mode, the clients only connect to one of the configured addresses.
+This single member behaves as a gateway to the other members.
+For any operation requested from the client, it redirects the request to the relevant member and
+returns the response back to the client returned from that member.
+
+Using the Unisocket mode is necessary in scenarios where clients are required to connect to only a single member rather than to each member in the cluster. This requirement may arise due to firewalls, security considerations, or specific networking issues.
+
+Since the {client-name} currently only supports the Unisocket mode, you need to disable Smart routing in the client network configuration as follows:
+
+[source,java]
+----
+ClientConfig clientConfig = new ClientConfig();
+ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
+networkConfig.setSmartRouting(false);
+----
+
+Note that you need also to disable Smart routing for the clients which
+want to use temporary permissions defined in a member.
+See the xref:security:native-client-security.adoc#handling-permissions-when-a-new-member-joins[Handling Permissions section].
+
+[[enabling-redo-operation]]
+==== Enabling Redo Operation
+
+It enables/disables redo-able operations as described in
+<<handling-failures, Handling Retry-able Operation Failure>>.
+The following is an example configuration.
+
+[source,java]
+----
+ClientConfig clientConfig = new ClientConfig();
+ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
+networkConfig().setRedoOperation(true);
+----
+
+Its default value is `false` (disabled).
+
+==== Setting a Socket Interceptor
+
+[blue]*Hazelcast {enterprise-product-name}*
+
+Following is a client configuration to set a socket intercepter.
```

Comment:
```suggestion
Following is a client configuration to set a socket interceptor.
```

pls revert any changes like this. It's called SocketInterceptor in Hazelcast terms

Reply:
Reverted, it was a typo